backend/src/main/java/com/team4/backend/controller/ReportController.java

```diff
@@ -19,16 +19,17 @@ public ReportController(ReportService reportService) {
         this.reportService = reportService;
     }
 
-    @GetMapping(value = "/generateAllNonValidatedOffersReport/{session}", produces = MediaType.APPLICATION_PDF_VALUE)
+
+    @GetMapping(value = "/generateAllNonValidatedOffersReport/{fullName}", produces = MediaType.APPLICATION_PDF_VALUE)
     @PreAuthorize("hasAuthority('INTERNSHIP_MANAGER')")
-    public Mono<byte[]> generateAllNonValidatedOffersReport(@PathVariable("session") Integer sessionNumber) {
-        return reportService.generateAllNonValidatedOffersReport(sessionNumber);
+    public Mono<byte[]> generateAllNonValidatedOffersReport(@PathVariable("fullName") String semesterFullName) {
```

Comment:
Maybe change variable name to `semesterFullName` in the path as well, not necessary just a remark